exercises/sam_jenkins/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <iomanip>
+#include <vector>
+#include "mapFunctions.h"
+
+
+using std::ifstream; using std::string; using std::cout; using std::endl; using std::istreambuf_iterator; 
+using std::unordered_map; using std::vector; using std::ofstream;
+
+
+int main(int argc, char **argv) {
+	//check for valid number of arguements
+	if (argc !=2) {
+		cout << "This Program requires a single input" << endl;
+			return 1;
+	}
+	//open the input stream
+	ifstream in(argv[1]);
+	if (in) {
+		//if the stream opened properly, read the entire file to a string, then make the string all lowercase.
+		string read((istreambuf_iterator<char>(in)), (istreambuf_iterator<char>()));
+		transform(read.begin(), read.end(), read.begin(), ::tolower);
+
+		//create an unordered map to count words
+		const unordered_map<string, int> count =countof(read);
```

Comment:
Use `auto` where you can. They type of count can be deduced from the function call. Same below.

Reply:
👍 that you're thinking of using `const` where you can.